source/Irrlicht/CBAWMeshFileLoader.cpp

```diff
@@ -208,10 +211,91 @@ void* CBAWMeshFileLoader::tryReadBlobOnStack(const SBlobData & _data, SContext &
 		dst = _stackPtr;
 	else
 		dst = malloc(_data.header->blobSizeDecompr);
+
+	const bool compressed = (_data.header->compressionType != core::Blob::EBCT_RAW);
+
+	uint8_t stackCompressed[1u<<14];
```

Comment:
Stack use here is redundant and possibly problematic.
1) We can't grab more stack safely (Windows has only 1MB by default)
2) Anything under 16k should not really be compressed
3) Statistically the larger decompressed stuff will be compressed

So either:
A) Don't try and make dst=_stackPtr and always malloc dst straight away and possibly use _stackPtr as working area for dstCompressed data
B) always malloc dstCompressed